### PR TITLE
feat: timeline encrypted tweet optimization

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
MF-4184

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/components/CompositionDialog/useSubmit.ts [hunk 1]
```diff
@@ -44,7 +44,9 @@ export function useSubmit(onClose: () => void, reason: 'timeline' | 'popup' | 'r
 
             if (encode === 'image') {
                 const decoratedText = decorateEncryptedText('', t, content.meta)
-                const defaultText = t('additional_post_box__encrypted_post_pre', { encrypted: '' })
+                const defaultText = t('additional_post_box__encrypted_post_pre', {
+                    encrypted: 'https://mask.io/',
```
Comment (Jack-Works) [inline packages/mask/src/components/CompositionDialog/useSubmit.ts:48]: you should use another string or variable name for it IMO.

File: packages/mask/src/components/InjectedComponents/PostReplacer.tsx [hunk 2]
```diff
@@ -28,7 +30,15 @@ export interface PostReplacerProps {
 
 export function PostReplacer(props: PostReplacerProps) {
     const { classes } = useStyles()
-    const postMessage = usePostInfoDetails.rawMessage()
+    const postMessage_ = usePostInfoDetails.rawMessage()
+    const messageAnchorWithPostData = postMessage_.items.find(
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/PostReplacer.tsx]: can you explain what you did here?

Comment (hanszhoou) [reply packages/mask/src/components/InjectedComponents/PostReplacer.tsx]: Hiding the  prompt text for users who have Mask installed.

File: packages/mask/src/components/CompositionDialog/useSubmit.ts (hunk 1 above)
Comment (hanszhoou) [reply packages/mask/src/components/CompositionDialog/useSubmit.ts:48]: I think it's proper. 

image

becomes

image

File: packages/mask/src/components/InjectedComponents/PostReplacer.tsx (hunk 2 above)
Comment (hanszhoou) [reply packages/mask/src/components/InjectedComponents/PostReplacer.tsx]: Replaced by `MaskMessages.events.postReplacerHidden`

**Review by guanbinrui — APPROVED** [review]

File: packages/mask/src/components/CompositionDialog/useSubmit.ts (hunk 1 above)
Comment (Jack-Works) [reply packages/mask/src/components/CompositionDialog/useSubmit.ts:48]: Facebook has different text for it. Please make sure it also looks good.

**Pull request merged**